### ISSUE TITLE
Look into why LH on WPT failures are occurring. 

### ISSUE THREAD

**Issue by vinamratasingal** [GoogleChrome/lighthouse#3719]
We've noticed a 20% failure on LH on WPT- let's look into why these failures are occurring.

**Comment by patrickhulce** [comment]
So it seems like the issue is the low-powered hardware being used for the mobile agents (raspberry pi's) causes Lighthouse to take more than the 4 minute (!!!!) timeout. It's hard to tell where this time is being spent right now other than it's not during page load. @paulirish's recent work to add the audit and gather times to the timing object should *really* help here.

URL | Lighthouse Time Taken | Load Time
-- | -- | --
http://www.fightingkids.com | 88s | 7s
http://www.city-data.com/ | 157s | 16s
http://www.locationsmobiltech. | 82s | 8s
http://www.satelectrodomestico | 157s | 16s
http://www.nadir.it | 88s | 9s
http://www.carccessory.com/cat | 115s | 13s
http://www.clz.com | 78s | 9s
http://www.priceminister.com/ | 188s | 22s
http://www.fc2web.com/ | 70s | 8s
http://www.radio.net.pk | 135s | 16s
http://www.orridge.eu | 73s | 9s
http://www.moreirajr.com.br | 110s | 13s
http://www.2ch.net/ | 113s | 14s
http://www.boosj.com/ | 137s | 17s
http://www.huanlang.com/ | 95s | 12s
http://www.brothersoft.com/ | 162s | 20s
http://www.dealtime.com/ | 117s | 15s
http://www.nhaccuatui.com/ | 150s | 19s
http://www.all-guitar-chords.c | 81s | 11s
http://www.bing.com/ | 81s | 11s
http://www.eyelashgarage.jp/ws | 182s | 24s
http://www.freewebs.com/ | 108s | 15s
http://www.webs.com/ | 104s | 14s
http://www.linwellparkdentalce | 119s | 16s
http://www.freedownloadscenter | 93s | 13s
http://www.novoteka.ru/ | 117s | 17s
http://www.95599.cn/ | 175s | 25s
http://www.unlimited-downloads | 99s | 14s
http://www.chip.de/ | 121s | 18s
http://www.6.cn/ | 146s | 22s
http://www.sanook.com/ | 139s | 20s
http://www.cnxad.com/ | 136s | 20s
http://www.robtex.com/ | 93s | 14s
http://www.barnesandnoble.com/ | 174s | 27s
http://www.renren.com/ | 86s | 13s
http://www.insto.co.id/eyeduca | 79s | 12s
http://www.japanpost.jp/ | 141s | 22s
http://www.guardian.co.uk/ | 142s | 23s
http://www.cooks.com/ | 79s | 13s
http://www.yfrog.com/ | 81s | 13s
http://www.jsoftj.com/ | 75s | 12s
http://www.swlgpc.com/jp/ | 143s | 23s
http://www.top-start.com/kateg | 96s | 15s
https://www.google.ca/search?c | 100s | 16s
http://www.imageshack.us/ | 77s | 13s
http://www.sciencedirect.com/ | 74s | 12s
http://www.liveinternet.ru/ | 101s | 16s
http://www.lyricsmode.com/ | 103s | 17s
http://www.xrea.com/ | 123s | 20s
https://tadaaz.nl/geboorte/lp/ | 134s | 22s
http://www.babylon.com/ | 87s | 15s
http://www.ft.com/ | 115s | 19s
http://www.lavante.com/solutio | 128s | 22s
http://www.metacafe.com/ | 121s | 20s
http://www.finyear.com | 82s | 14s
http://www.baidu.com/ | 78s | 13s
http://www.kongres-content.pl/ | 154s | 26s
http://www.dell.com/ | 84s | 14s
http://www.satokazzz.com | 76s | 13s
http://www.allocine.fr/ | 124s | 21s
http://www.teambuildingsolutio | 95s | 17s
http://injury.thejusticeattorn | 91s | 16s
http://www.findarticles.com/ | 75s | 13s
http://www.babycenter.com/ | 98s | 17s
http://www.lyrics007.com/ | 83s | 15s
http://www.terra.com.br/ | 158s | 28s
http://www.autoscout24.de/ | 108s | 19s
http://www.kitwiboon.com/15397 | 91s | 17s
http://www.univision.com/ | 183s | 34s
http://m.sdi.bg/page/10231.htm | 87s | 16s
http://www.aufeminin.com/ | 109s | 20s
http://www.corriere.it/ | 98s | 18s
https://www.farmingparts.com/M | 84s | 16s
http://www.ca.gov/ | 163s | 31s
http://www.softpedia.com/ | 80s | 15s
http://www.amazon.co.jp/ | 88s | 17s
http://www.movies-links.tv/ | 103s | 20s
http://www.popeater.com/ | 156s | 30s
http://www.quranexplorer.com | 131s | 25s
http://www.gmx.net/ | 114s | 22s
http://www.samsung.com/ | 91s | 18s
http://www.expedia.com/ | 163s | 32s
http://www.manta.com/ | 94s | 19s
http://www.bild.de/ | 194s | 40s
http://www.vimeo.com/ | 95s | 20s
http://www.amazon.de/ | 98s | 20s
http://www.amazon.fr/ | 99s | 20s
http://www.rockyou.com/ | 149s | 31s
http://www.sears.com/ | 151s | 32s
http://modpodpodiatry.com.au/l | 91s | 20s
http://www.marketwatch.com/ | 141s | 31s
http://irlandaingilizce.com/ir | 115s | 25s
http://www.bayshorefurniture.c | 89s | 20s
http://www.meister-farben.de/w | 139s | 31s
http://services.martellaelectr | 102s | 23s
http://www.gentosha-book.com/l | 96s | 21s
http://www.amazon.co.uk/ | 102s | 23s
http://www.abby-life.com/shopb | 93s | 21s
http://www.excite.co.jp/ | 108s | 24s
http://www.rcgp.org.uk | 86s | 19s
http://www.lowes.com/ | 92s | 21s
http://www.pixnet.net/ | 110s | 26s
http://www.sanspo.com/ | 153s | 37s
http://www.yomiuri.co.jp/ | 131s | 32s
http://www.mapion.co.jp/ | 99s | 24s
http://www.altcointoday.com | 115s | 28s
http://www.pchome.com.tw/ | 120s | 30s
http://www.istockphoto.com/ | 103s | 26s
http://xtremewhitetailadventur | 175s | 44s
http://www.orbitz.com/ | 107s | 28s
https://tsunagaru-kusw.jp/educ | 99s | 26s
http://www.legacy.com/ | 103s | 27s
http://www.amazon.com/ | 117s | 31s
http://www.baijingapp.com | 176s | 46s
http://www.adaptnetwork.com | 122s | 33s
http://gretaflowers.pl/kategor | 119s | 32s
http://www.dress-for-less.com | 88s | 24s
https://roboshkola.com/catalog | 95s | 26s
http://www.myvideo.de/ | 94s | 26s
https://www.printworks.co.jp/m | 124s | 36s
http://www.hatena.ne.jp/ | 142s | 42s
https://www.europeansnowsport. | 185s | 55s
http://www.hulu.com/ | 88s | 27s
http://m.bosch-do-it.de/de/de/ | 129s | 44s
http://www.reviewr.com | 119s | 45s
http://www.dropbox.com/ | 142s | 54s
http://www.urbandictionary.com | 146s | 55s
http://www.bebo.com/ | 189s | 86s
http://www.pagesjaunes.fr/ | 210s | 120s

**Comment by patrickhulce** [comment]
Alright a few more useful nuggets. The biggest problem seems to be sites that never reach CPU quiet and make it impossible to capture. 

Example: www.tiancity.com/

* A long task occurs every second *forever*
* Lighthouse will trace the entire timeout length (120s on WPT)
* One of two things happen
  * A) Chrome kills the process because a Moto G4 doesn't have enough RAM to hold a trace thats *two minutes* long
  * B) Retrieving the enormous trace takes up our remaining time and WPT kills Lighthouse for taking too long

**Comment by patrickhulce** [comment]
OK, I re-ran the same set of 100 and we're down to a ~12% failure rate now after reducing the timeout to 80s 👍 

Digging into the remaining errors there are...

* 8 sites that crash Chrome on the phone when trying to retrieve the trace, investigating how low we need to bring the timeout to fix these
* 4 chinese sites whose servers sometimes timeout, re-running the test fixed the issue
* 1 site that still needed more time, loading + retrieving trace took ~3 minutes, gatherers took ~45s, pSI took ~60s which exceeds the 4 minute limit

In short, it looks like there's very little that's actionable from the LH side. We can try to improve things for that 1 site, but then again if 10% of sites still crash because we're waiting too long, bringing down the timeout will likely fix this site anyway.

**Comment by vinamratasingal** [comment]
Hey Patrick- QQ. When you say that we're down to a 12% error rate, is that from 20? Are the 100 sites you tested on sites that gave us errors, or just a representative subset of the 1500?

**Comment by patrickhulce** [comment]
@vinamratasingal yeah its just a random sample of the 1500 that I reran, it's 12% error rate down from 20% so still a very significant percentage of sites

**Comment by patrickhulce** [comment]
Just updated the failures with necessary timeout information. Given the vast majority of failures are beyond our immediate control and progress on that effort is tracked by #3596, I'm going to go ahead and close this issue. 

Possible speed improvements have been filed into respective repositories and can be found below if we solve #3596 such that they become bottlenecks.
* [ ] #3596
* [ ] #3797
* [ ] https://github.com/paulirish/speedline/issues/29
* [ ] https://github.com/paulirish/speedline/issues/57
* [ ] https://github.com/paulirish/speedline/issues/58
